### PR TITLE
fix(docs, http_server source): Update docs for disabling `max_connection_age_secs`

### PR DESCRIPTION
**Pull request by jszwedko**
Ideally you could set this to `null`, but Vector requires all config values to be valid TOML, of
which `null` is not.

Closes: https://github.com/vectordotdev/vector/issues/19644

Signed-off-by: Jesse Szwedko <jesse.szwedko@datadoghq.com>

### CODE REVIEW

File: src/http.rs [hunk 1]
```diff
@@ -401,6 +401,10 @@ pub struct KeepaliveConfig {
     /// The maximum amount of time a connection may exist before it is closed
     /// by sending a `Connection: close` header on the HTTP response.
     ///
+    ///
+    /// Set this to a high number to disable this feature (in the future setting this to `null`
```
Comment (neuronull) [inline src/http.rs:405]:
```suggestion
    /// Set this to a large value to disable this feature (in the future setting this to `null`
```

Or `large number` ?

Also wondering if an example of the large number would be useful.

**Review by neuronull — APPROVED** [review]

File: website/content/en/highlights/2023-12-19-0-35-0-upgrade-guide.md [hunk 2]
```diff
@@ -83,7 +83,9 @@ interpolation.
 HTTP server-based sources include a new `keepalive.max_connection_age_secs` configuration option, which defaults to 5 minutes (300 seconds).
 When enabled, this closes incoming TCP connections that reach the maximum age by sending a `Connection: close` header in the response.
 While this parameter is crucial for managing the lifespan of persistent, incoming connections to Vector and for effective load balancing, it
-can be disabled by setting `keepalive.max_connection_age_secs` to `null`.
+can be disabled by setting `keepalive.max_connection_age_secs` to a very high number (ideally you
```
Comment (jszwedko) [inline website/content/en/highlights/2023-12-19-0-35-0-upgrade-guide.md:86]:
```suggestion
can be disabled by setting `keepalive.max_connection_age_secs` to a very large number (ideally you
```

File: src/http.rs [hunk 3]
```diff
@@ -401,6 +401,10 @@ pub struct KeepaliveConfig {
     /// The maximum amount of time a connection may exist before it is closed
     /// by sending a `Connection: close` header on the HTTP response.
     ///
+    ///
+    /// Set this to a large value to disable this feature (in the future setting this to `null`
```
Comment (jszwedko) [inline src/http.rs:405]:
```suggestion
    /// Set this to a large value like `10000000` to disable this feature (in the future setting this to `null`
```

File: src/http.rs (hunk 1 above)
Comment (jszwedko) [reply src/http.rs]: Good call, `large` is better than `high` 😄 I'll add an example too.

File: src/http.rs [hunk 4]
```diff
@@ -401,6 +401,10 @@ pub struct KeepaliveConfig {
     /// The maximum amount of time a connection may exist before it is closed
     /// by sending a `Connection: close` header on the HTTP response.
     ///
+    ///
+    /// Set this to a large value like `100000000` to disable this feature (in the future setting
```
Comment (hestonhoffman) [inline src/http.rs:405]:
```suggestion
    /// Set this to a large value like `100000000` to disable this feature (in the future, setting
```
We generally avoid these kinds of "future" statements in the docs.

File: website/content/en/highlights/2023-12-19-0-35-0-upgrade-guide.md [hunk 5]
```diff
@@ -83,7 +83,9 @@ interpolation.
 HTTP server-based sources include a new `keepalive.max_connection_age_secs` configuration option, which defaults to 5 minutes (300 seconds).
 When enabled, this closes incoming TCP connections that reach the maximum age by sending a `Connection: close` header in the response.
 While this parameter is crucial for managing the lifespan of persistent, incoming connections to Vector and for effective load balancing, it
-can be disabled by setting `keepalive.max_connection_age_secs` to `null`.
+can be disabled by setting `keepalive.max_connection_age_secs` to a very large number like
+`100000000` (ideally you could set it to `null` but Vector currently requires all values to be valid
```
Comment (hestonhoffman) [inline website/content/en/highlights/2023-12-19-0-35-0-upgrade-guide.md:87]:
```suggestion
`100000000` (ideally you could set it to `null`, but Vector currently requires all values to be valid
```

**Review by hestonhoffman — APPROVED** [review]

File: src/http.rs (hunk 4 above)
Comment (jszwedko) [reply src/http.rs]: Thanks, agreed it does seem prudent to avoid these sorts of future statements since they may never come to be.

**Pull request merged**